SerialPrograms/Source/PokemonSV/Programs/PokemonSV_Navigation.cpp

```diff
@@ -398,6 +399,120 @@ void leave_box_system_to_overworld(const ProgramInfo& info, ConsoleHandle& conso
 }
 
 
+void open_pokedex_from_overworld(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context){
+    {
+        OverworldWatcher overworld(COLOR_CYAN);
+        context.wait_for_all_requests();
+        int ret = wait_until(
+            console, context,
+            std::chrono::seconds(10),
+            {overworld}
+        );
+        context.wait_for(std::chrono::milliseconds(100));
+        if (ret == 0){
+            console.log("Detected overworld.");
+            pbf_press_button(context, BUTTON_MINUS, 20, 105); // open pokedex
+        }
+        else{
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "open_pokedex_from_overworld(): No overworld state found after 10 seconds.",
+                true
+            );
+        }
+    }
+
+    WallClock start = current_time();
+    while (true){
+        if (current_time() - start > std::chrono::minutes(1)){

```

Comment:
Could that time be reduced? 1 minute sounds huge

Reply:
Reduced to 30 seconds